server/middleware/validateRequest.js

```diff
@@ -0,0 +1,68 @@
+var jwt = require('jwt-simple');
+var validateUser = require('../routes/auth').validateUser;
+ 
+module.exports = function(req, res, next) {
+ 
+  // When performing a cross domain request, you will recieve
+  // a preflighted request first. This is to check if our the app
+  // is safe.  
+ 
+  var token = (req.body && req.body.access_token) || (req.query && req.query.access_token) || req.headers['x-access-token'];
+  var key = (req.body && req.body.x_key) || (req.query && req.query.x_key) || req.headers['x-key'];
+ 
+  if (token || key) {
+    try {
+      var decoded = jwt.decode(token, require('../config/secret.js')());
+ 
+      if (decoded.exp <= Date.now()) {
+        res.status(400);
+        res.json({
+          "status": 400,
+          "message": "Token Expired"
```

Comment:
Duas coisas aqui que acho que poderiam ser mudadas. 
1- Mudar o status. Acho que 400 fica muito generico e mais dificil de lidar no frontend.
2- Consertar isso no front. Como existe a possibilidade do token expirar, isto deveria ser tratado no frontend. Já que o token fica alocado lá no localStorage que não tem prazo definido para expirar.

Para isso vejo duas alternativas:
1 - Mais simples e talvez menos coesa: https://stackoverflow.com/questions/2326943/when-do-items-in-html5-local-storage-expire. Ou seja, tentando bater o timestamp do token do servidor com o localStorage(o que não acho muito correto, mas, é simples)
2 - Criando interceptor. Aqui tem um link de como fazer e que casa bem com nossa aplicação:
https://medium.com/@ryanchenkie_40935/angular-authentication-using-the-http-client-and-http-interceptors-2f9d1540eb8